### PR TITLE
updated to latest yeoman-generator (0.22)

### LINKED ISSUE #42: Base on yeoman-generator v0.18
Since v2.0.0 is a breaking change, might as well be on latest ver of yeoman-generator. Also, there is a very *slight* chance that we'll need this for #39. 

Even if not, I'm almost certain that we'll want it for v2.1 so the default generator can store the user's ES6 preference for later use in the widget generator. So if this does not happen for v2.0 it will go into v2.1.

See the [v0.18 blog post](http://yeoman.io/blog/release-0.18.html) for info.

### PR DESCRIPTION
**Pull request by tomwayson**
Updated to use latest version of yeoman-generator (0.22).

Minor changes to generator files but a major overhaul to test files.

Resolves #42 and resolves #35.

All tests passing, and tested output against local install of WAB, and seems to be working well.

@gavinr please pull and test locally. Will require `npm install`.

### CODE REVIEW

File: widget/index.js [hunk 1]
```diff
@@ -149,7 +149,7 @@ var WidgetGenerator = yeoman.generators.Base.extend({
   files: function () {
     // NOTE: this is needed b/c _Widget.html has ES6 style interpolation delimiters
     // see: https://github.com/lodash/lodash/issues/399
-    this._.templateSettings.interpolate = /<%=([\s\S]+?)%>/g;
+    // this._.templateSettings.interpolate = /<%=([\s\S]+?)%>/g;
```
Comment (tomwayson) [inline widget/index.js]: I should have removed this instead of commented out

**Comment by tomwayson** [comment]
Looks like travis fails only on node v.10, which raises the question, what is the minimum version of node (and npm?) should this support? Especially since this is a major (breaking change) release. Will have to look into yeoman-generator and see, but [at first glance it seems like v.0.10 should be OK](https://github.com/yeoman/generator/blob/v0.22.5/package.json#L27-L29).

**Comment by gavinr** [comment]
I have tested the full workflow and seems to work well on my side. Nice work! 

I think it sounds reasonable to drop v.10 support.

**Comment by tomwayson** [comment]
Well, I've tried it locally w/ nvm and same prob w/ v0.12. So prob worth spending some more time on figuring out the root causes.

**Comment by gavinr** [comment]
:+1:

**Pull request merged**